docs/timezone.md

```diff
@@ -1,19 +1,213 @@
 # `Temporal.TimeZone`
 
-A representation of a time-zone giving access to information about offsets & dst-changes as well as allowing for conversions.
+A `Temporal.TimeZone` is a representation of a time zone: either an [IANA time zone](https://www.iana.org/time-zones), including information about the time zone such as the offset between the local time and UTC at a particular time, and daylight saving time (DST) changes; or simply a particular UTC offset with no DST.
 
-## new Temporal.TimeZone(timeZone: string) : Temporal.TimeZone
+Since `Temporal.Absolute` and `Temporal.DateTime` do not contain any time zone information, a `Temporal.TimeZone` object is required to convert between the two.
+
+Finally, the `Temporal.TimeZone` object itself provides access to a list of the time zones in the IANA time zone database.
+
+## new Temporal.TimeZone(timeZoneIdentifier: string) : Temporal.TimeZone
+
+**Parameters:**
+- `timeZoneIdentifier` (string): A description of the time zone; either its IANA name, or a UTC offset.
+
+**Returns:** a new `Temporal.TimeZone` object.
+
+For a list of IANA time zone names, see the current version of the [IANA time zone database](https://www.iana.org/time-zones).
+A convenient list is also available [on Wikipedia](https://en.wikipedia.org/wiki/List_of_tz_database_time_zones), although it might not reflect the latest official status.
+
+The string `timeZoneIdentifier` is canonicalized before being used to determine the time zone.
+For example, values like `+100` will be understood to mean `+01:00`, and capitalization will be corrected.
+If no time zone can be determined from `timeZoneIdentifier`, then a `RangeError` is thrown.
+
+Usage examples:
+```javascript
+tz = new Temporal.TimeZone('UTC');
+tz = new Temporal.TimeZone('Africa/Cairo');
+tz = new Temporal.TimeZone('america/VANCOUVER');
+tz = new Temporal.TimeZone('Asia/Katmandu');  // alias of Asia/Kathmandu
+tz = new Temporal.TimeZone('-04:00');
+tz = new Temporal.TimeZone('+645');
+// tz = new Temporal.TimeZone('local');  // not a time zone, throws
```

Comment:
Not necessarily, I thought it would be useful to show that it's incorrect so that people don't copy-paste it, but if that's confusing then probably the "throws" comment at the end is enough. Or we could do something like this:
```js
/* WRONG */ tz = new Temporal.TimeZone('local');  // not a time zone, throws
```

Reply:
I went with the latter.

Reply:
Thanks.